utils/utils.c

```diff
@@ -192,18 +192,23 @@ static bool is_uftrace_directory(const char *path)
 	fd = open(info_path, O_RDONLY);
 	free(info_path);
 
-	if (fd == -1)
-		return false;
-
-	if (read(fd, sig, UFTRACE_MAGIC_LEN) != UFTRACE_MAGIC_LEN) {
-		/*
-		 * partial read() will return false anyway
-		 * since memcmp() below cannot success.
-		 */
+	if (fd != -1) {
+		if (read(fd, sig, UFTRACE_MAGIC_LEN) != UFTRACE_MAGIC_LEN) {
+			/*
+			 * partial read() will return false anyway
+			 * since memcmp() below cannot success.
+			 */
+		}
+		close(fd);
+		return !memcmp(sig, UFTRACE_MAGIC_STR, UFTRACE_MAGIC_LEN);
 	}
 
-	close(fd);
-	return !memcmp(sig, UFTRACE_MAGIC_STR, UFTRACE_MAGIC_LEN);
+	/* if "info" file is missing, also check that there is "default.opts" */
+	xasprintf(&info_path, "%s/default.opts", path);
+	if (!access(info_path, F_OK))
+		return true;
+
+	return false;
```

Comment:
You need to free `info_path` before the return.

Reply:
I missed it. Thanks.